### ISSUE TITLE
undefined reference to `gotoblas' Makefile:406: recipe for target 'bin/im2rec' failed

### ISSUE THREAD

**Issue by DevinCheung** [apache/incubator-mxnet#8773]
cd /dev/shm/multitask/mx-maskrcnn/incubator-mxnet/dmlc-core; make libdmlc.a USE_SSE=1 config=/dev/shm/multitask/mx-maskrcnn/incubator-mxnet/make/config.mk; cd /dev/shm/multitask/mx-maskrcnn/incubator-mxnet
make[1]: Entering directory '/dev/shm/multitask/mx-maskrcnn/incubator-mxnet/dmlc-core'
make[1]: 'libdmlc.a' is up to date.
make[1]: Leaving directory '/dev/shm/multitask/mx-maskrcnn/incubator-mxnet/dmlc-core'
g++ -DMSHADOW_FORCE_STREAM -Wall -Wsign-compare -O3 -DNDEBUG=1 -I/dev/shm/multitask/mx-maskrcnn/incubator-mxnet/mshadow/ -I/dev/shm/multitask/mx-maskrcnn/incubator-mxnet/dmlc-core/include -fPIC -I/dev/shm/multitask/mx-maskrcnn/incubator-mxnet/nnvm/include -I/dev/shm/multitask/mx-maskrcnn/incubator-mxnet/dlpack/include -Iinclude -funroll-loops -Wno-unused-variable -Wno-unused-parameter -Wno-unknown-pragmas -Wno-unused-local-typedefs -msse3 -I/usr/local/cuda/include -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_MKL=0 -DMSHADOW_RABIT_PS=0 -DMSHADOW_DIST_PS=0 -DMSHADOW_USE_PASCAL=0 -DMXNET_USE_OPENCV=1 -I/usr/include/opencv -fopenmp -DMXNET_USE_LAPACK -DMSHADOW_USE_CUDNN=1 -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free  -I/dev/shm/multitask/mx-maskrcnn/incubator-mxnet/cub -DMXNET_USE_LIBJPEG_TURBO=0 -std=c++11  -o bin/im2rec tools/im2rec.cc build/src/operator/mkl/mkl_cppwrapper.o build/src/operator/mkl/mkl_memory.o build/src/operator/random/sample_multinomial_op.o build/src/operator/random/multisample_op.o build/src/operator/random/sample_op.o build/src/operator/tensor/elemwise_binary_broadcast_op_extended.o build/src/operator/tensor/elemwise_binary_op_logic.o build/src/operator/tensor/elemwise_binary_op_extended.o build/src/operator/tensor/square_sum.o build/src/operator/tensor/dot.o build/src/operator/tensor/elemwise_sum.o build/src/operator/tensor/init_op.o build/src/operator/tensor/elemwise_binary_broadcast_op_basic.o build/src/operator/tensor/cast_storage.o build/src/operator/tensor/elemwise_binary_op.o build/src/operator/tensor/elemwise_binary_scalar_op_logic.o build/src/operator/tensor/elemwise_scatter_op.o build/src/operator/tensor/elemwise_unary_op_basic.o build/src/operator/tensor/broadcast_reduce_op_value.o build/src/operator/tensor/ordering_op.o build/src/operator/tensor/elemwise_binary_op_basic.o build/src/operator/tensor/elemwise_binary_scalar_op_basic.o build/src/operator/tensor/indexing_op.o build/src/operator/tensor/elemwise_binary_broadcast_op_logic.o build/src/operator/tensor/la_op.o build/src/operator/tensor/broadcast_reduce_op_index.o build/src/operator/tensor/sparse_retain.o build/src/operator/tensor/control_flow_op.o build/src/operator/tensor/elemwise_binary_scalar_op_extended.o build/src/operator/tensor/matrix_op.o build/src/operator/tensor/elemwise_unary_op_trig.o build/src/operator/nnpack/nnpack_util.o build/src/operator/contrib/multibox_target.o build/src/operator/contrib/proposal.o build/src/operator/contrib/count_sketch.o build/src/operator/contrib/dequantize.o build/src/operator/contrib/deformable_psroi_pooling.o build/src/operator/contrib/fft.o build/src/operator/contrib/multibox_prior.o build/src/operator/contrib/ctc_loss.o build/src/operator/contrib/multi_proposal.o build/src/operator/contrib/psroi_pooling.o build/src/operator/contrib/quantize.o build/src/operator/contrib/deformable_convolution.o build/src/operator/contrib/ifft.o build/src/operator/contrib/multibox_detection.o build/src/operator/custom/native_op.o build/src/operator/custom/ndarray_op.o build/src/operator/custom/custom.o build/src/operator/nn/softmax.o build/src/io/image_aug_default.o build/src/io/io.o build/src/io/iter_csv.o build/src/io/iter_image_det_recordio.o build/src/io/image_io.o build/src/io/image_det_aug_default.o build/src/io/iter_image_recordio.o build/src/io/iter_mnist.o build/src/io/iter_image_recordio_2.o build/src/io/iter_libsvm.o build/src/common/rtc.o build/src/common/utils.o build/src/nnvm/legacy_op_util.o build/src/nnvm/legacy_json_util.o build/src/imperative/cached_op.o build/src/imperative/imperative.o build/src/ndarray/ndarray_function.o build/src/ndarray/ndarray.o build/src/operator/instance_norm.o build/src/operator/pooling.o build/src/operator/crop.o build/src/operator/spatial_transformer.o build/src/operator/swapaxis.o build/src/operator/convolution_v1.o build/src/operator/pad.o build/src/operator/batch_norm.o build/src/operator/softmax_output.o build/src/operator/cudnn_algoreg.o build/src/operator/correlation.o build/src/operator/operator_util.o build/src/operator/sequence_reverse.o build/src/operator/bilinear_sampler.o build/src/operator/sequence_last.o build/src/operator/svm_output.o build/src/operator/operator.o build/src/operator/optimizer_op.o build/src/operator/lrn.o build/src/operator/convolution.o build/src/operator/fully_connected.o build/src/operator/batch_norm_v1.o build/src/operator/sequence_mask.o build/src/operator/concat.o build/src/operator/grid_generator.o build/src/operator/pooling_v1.o build/src/operator/leaky_relu.o build/src/operator/identity_attach_KL_sparse_reg.o build/src/operator/deconvolution.o build/src/operator/roi_pooling.o build/src/operator/cudnn_batch_norm.o build/src/operator/loss_binary_op.o build/src/operator/regression_output.o build/src/operator/make_loss.o build/src/operator/upsampling.o build/src/operator/rnn.o build/src/operator/roi_align.o build/src/operator/slice_channel.o build/src/operator/dropout.o build/src/operator/cross_device_copy.o build/src/operator/softmax_activation.o build/src/operator/activation.o build/src/operator/l2_normalization.o build/src/engine/profiler.o build/src/engine/naive_engine.o build/src/engine/openmp.o build/src/engine/threaded_engine_pooled.o build/src/engine/engine.o build/src/engine/threaded_engine.o build/src/engine/threaded_engine_perdevice.o build/src/storage/storage.o build/src/c_api/c_api_symbolic.o build/src/c_api/c_api_ndarray.o build/src/c_api/c_api_executor.o build/src/c_api/c_predict_api.o build/src/c_api/c_api_function.o build/src/c_api/c_api.o build/src/c_api/c_api_error.o build/src/executor/inplace_addto_detect_pass.o build/src/executor/graph_executor.o build/src/executor/infer_graph_attr_pass.o build/src/executor/attach_op_execs_pass.o build/src/executor/attach_op_resource_pass.o build/src/kvstore/kvstore.o build/src/resource.o build/src/initialize.o /dev/shm/multitask/mx-maskrcnn/incubator-mxnet/dmlc-core/libdmlc.a /dev/shm/multitask/mx-maskrcnn/incubator-mxnet/nnvm/lib/libnnvm.a build/src/operator/random/sample_multinomial_op_gpu.o build/src/operator/random/multisample_op_gpu.o build/src/operator/random/sample_op_gpu.o build/src/operator/tensor/indexing_op_gpu.o build/src/operator/tensor/elemwise_binary_scalar_op_extended_gpu.o build/src/operator/tensor/elemwise_binary_scalar_op_basic_gpu.o build/src/operator/tensor/matrix_op_gpu.o build/src/operator/tensor/elemwise_unary_op_trig_gpu.o build/src/operator/tensor/elemwise_binary_op_extended_gpu.o build/src/operator/tensor/elemwise_sum_gpu.o build/src/operator/tensor/init_op_gpu.o build/src/operator/tensor/cast_storage_gpu.o build/src/operator/tensor/elemwise_binary_scalar_op_logic_gpu.o build/src/operator/tensor/dot_gpu.o build/src/operator/tensor/elemwise_unary_op_basic_gpu.o build/src/operator/tensor/ordering_op_gpu.o build/src/operator/tensor/broadcast_reduce_op_value_gpu.o build/src/operator/tensor/elemwise_binary_broadcast_op_basic_gpu.o build/src/operator/tensor/elemwise_binary_op_basic_gpu.o build/src/operator/tensor/elemwise_binary_broadcast_op_extended_gpu.o build/src/operator/tensor/broadcast_reduce_op_index_gpu.o build/src/operator/tensor/elemwise_scatter_op_gpu.o build/src/operator/tensor/elemwise_binary_broadcast_op_logic_gpu.o build/src/operator/tensor/control_flow_op_gpu.o build/src/operator/tensor/la_op_gpu.o build/src/operator/tensor/elemwise_binary_op_logic_gpu.o build/src/operator/tensor/sparse_retain_gpu.o build/src/operator/contrib/ctc_loss_gpu.o build/src/operator/contrib/psroi_pooling_gpu.o build/src/operator/contrib/quantize_gpu.o build/src/operator/contrib/deformable_convolution_gpu.o build/src/operator/contrib/ifft_gpu.o build/src/operator/contrib/multibox_detection_gpu.o build/src/operator/contrib/multibox_target_gpu.o build/src/operator/contrib/proposal_gpu.o build/src/operator/contrib/count_sketch_gpu.o build/src/operator/contrib/dequantize_gpu.o build/src/operator/contrib/deformable_psroi_pooling_gpu.o build/src/operator/contrib/fft_gpu.o build/src/operator/contrib/multibox_prior_gpu.o build/src/operator/contrib/multi_proposal_gpu.o build/src/operator/custom/native_op_gpu.o build/src/operator/nn/softmax_gpu.o build/src/common/utils_gpu.o build/src/ndarray/ndarray_function_gpu.o build/src/operator/optimizer_op_gpu.o build/src/operator/lrn_gpu.o build/src/operator/instance_norm_gpu.o build/src/operator/pooling_v1_gpu.o build/src/operator/fully_connected_gpu.o build/src/operator/sequence_mask_gpu.o build/src/operator/swapaxis_gpu.o build/src/operator/softmax_activation_gpu.o build/src/operator/pooling_gpu.o build/src/operator/leaky_relu_gpu.o build/src/operator/identity_attach_KL_sparse_reg_gpu.o build/src/operator/activation_gpu.o build/src/operator/batch_norm_v1_gpu.o build/src/operator/roi_pooling_gpu.o build/src/operator/cudnn_batch_norm_gpu.o build/src/operator/loss_binary_op_gpu.o build/src/operator/regression_output_gpu.o build/src/operator/l2_normalization_gpu.o build/src/operator/batch_norm_gpu.o build/src/operator/upsampling_gpu.o build/src/operator/roi_align_gpu.o build/src/operator/slice_channel_gpu.o build/src/operator/dropout_gpu.o build/src/operator/svm_output_gpu.o build/src/operator/pad_gpu.o build/src/operator/deconvolution_gpu.o build/src/operator/correlation_gpu.o build/src/operator/make_loss_gpu.o build/src/operator/grid_generator_gpu.o build/src/operator/softmax_output_gpu.o build/src/operator/rnn_gpu.o build/src/operator/convolution_v1_gpu.o build/src/operator/crop_gpu.o build/src/operator/sequence_reverse_gpu.o build/src/operator/spatial_transformer_gpu.o build/src/operator/bilinear_sampler_gpu.o build/src/operator/concat_gpu.o build/src/operator/sequence_last_gpu.o build/src/operator/convolution_gpu.o -pthread -lm -lcudart -lcublas -lcurand -lcusolver -L/usr/local/cuda/lib64 -L/usr/local/cuda/lib -lblas -fopenmp -lrt /usr/lib/x86_64-linux-gnu/libopencv_calib3d.so -lopencv_calib3d /usr/lib/x86_64-linux-gnu/libopencv_contrib.so -lopencv_contrib /usr/lib/x86_64-linux-gnu/libopencv_core.so -lopencv_core /usr/lib/x86_64-linux-gnu/libopencv_features2d.so -lopencv_features2d /usr/lib/x86_64-linux-gnu/libopencv_flann.so -lopencv_flann /usr/lib/x86_64-linux-gnu/libopencv_gpu.so -lopencv_gpu /usr/lib/x86_64-linux-gnu/libopencv_highgui.so -lopencv_highgui /usr/lib/x86_64-linux-gnu/libopencv_imgproc.so -lopencv_imgproc /usr/lib/x86_64-linux-gnu/libopencv_legacy.so -lopencv_legacy /usr/lib/x86_64-linux-gnu/libopencv_ml.so -lopencv_ml /usr/lib/x86_64-linux-gnu/libopencv_objdetect.so -lopencv_objdetect /usr/lib/x86_64-linux-gnu/libopencv_ocl.so -lopencv_ocl /usr/lib/x86_64-linux-gnu/libopencv_photo.so -lopencv_photo /usr/lib/x86_64-linux-gnu/libopencv_stitching.so -lopencv_stitching /usr/lib/x86_64-linux-gnu/libopencv_superres.so -lopencv_superres /usr/lib/x86_64-linux-gnu/libopencv_ts.so /usr/lib/x86_64-linux-gnu/libopencv_video.so -lopencv_video /usr/lib/x86_64-linux-gnu/libopencv_videostab.so -lopencv_videostab -llapack -lcudnn /usr/lib/libtcmalloc.so  -lcuda -lcufft -lnvrtc
/usr/lib/gcc/x86_64-linux-gnu/5/../../../../lib/libblas.so: undefined reference to `gotoblas'
collect2: error: ld returned 1 exit status
Makefile:406: recipe for target 'bin/im2rec' failed
make: *** [bin/im2rec] Error 1

I met the above error when I build mxnet from source. Who can help me?

**Comment by DevinCheung** [comment]
And I find the following about libblas in /usr/lib/:
libblas
libblas.a
libblas.so
libblas.so.3
libblas.so.3gf

**Comment by b0noI** [comment]
proposed labels: "Build", "C++", "Ubuntu"

**Comment by kpmurali** [comment]
I had the same issue. 

I got it fixed by following the suggestions of @staticfloat and @xiaoweiz in this earlier issue

https://github.com/JuliaLang/julia/issues/11913

I also did the following:

./ci/docker/install/ubuntu_core.sh
./ci/docker/install/ubuntu_python.sh
./ci/docker/install/ubuntu_scala.sh
./ci/docker/install/ubuntu_docs.sh

git submodule update --recursive
git submodule init
make -j16
make docs

**Comment by aaronmarkham** [comment]
I keep running into this issue as well. A clean clone with recursive init of the submodules seems to get around it, but this isn't a real fix.
